hosting/libs/firebase/auth.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  type User,
+  GoogleAuthProvider,
+  signInWithPopup,
+  onAuthStateChanged as _onAuthStateChanged,
+} from "firebase/auth";
+
+import { firebaseAuth } from "./config";
+
+export function onAuthStateChanged(callback: (authUser: User | null) => void) {
+  return _onAuthStateChanged(firebaseAuth, callback);
+}
+
+export async function signInWithGoogle() {
+  const provider = new GoogleAuthProvider();
+
+  try {
+    const result = await signInWithPopup(firebaseAuth, provider);
+
+    if (!result || !result.user) {
+      throw new Error("Google sign in failed");
+    }
+    return result.user.uid;
+  } catch (error) {
+    console.error("Error signing in with Google", error);
```

Comment:
It would be better if we could display information to the user with a popup message when an error occurs and display an error message in the browser console just for developer debugging.
So the user can find out whether the process was a success or an error
```suggestion
    Snackbar.error(new Error('Login failed'))
    console.error(error);
```

Reply:
Whenever possible, make an error service that wrap the error UI behavior and console log, so that we only call one method and can do customization in one place.